### ISSUE TITLE
rule 'no-unused-vars' not correctly applied

### ISSUE THREAD

**Issue by peduxe** [eslint/eslint#3417]
Reports `no-unused-vars` for the variable `notOk` running eslint from the terminal with the following [eslint resource file](https://gist.github.com/peduxe/fbba0d9aa5328bd14962).

note: this behavior doesn't happen with imports:
```js
import lib from 'lib';

lib.x = 5;
// or
lib.do();
```

target code:
```js
let ok;

if (ok === true) {
  ok = 5;
}

let notOk = 5;

notOk = 12;

if (true) {
  notOk = 12;
}
```

Output: `teste.js 7:5 warning notOk is defined but never used no-unused-vars ✖ 1 problem (0 errors, 1 warning)`

babel-eslint@4.0.7
eslint@1.1.0

**Comment by eslintbot** [comment]
Thanks for the issue! We get a lot of issues, so this message is automatically posted to each one to help you check that you've included all of the information we need to help you.

Reporting a bug? Please be sure to include:

1. The version of ESLint you are using (run `eslint -v`)
1. The source code that caused the problem
1. The configuration you're using (for the rule or your entire config file)
1. The actual ESLint output complete with line numbers

Requesting a new rule? Please be sure to include:

1. The use case for the rule - what is it trying to prevent or flag?
1. Whether the rule is trying to prevent an error or is purely stylistic
1. Why you believe this rule is generic enough to be included

Requesting a feature? Please be sure to include:

1. The problem you want to solve (don't mention the solution)
1. Your take on the correct solution to problem

Including this information in your issue helps us to triage it and get you a response as quickly as possible.

Thanks!

**Comment by michaelficarra** [comment]
Not a bug. It's never read.

**Comment by peduxe** [comment]
@michaelficarra Didn't say it is a bug though, by "**it's never read**", you mean even if I change the warning level it doesn't check against the rule? No, that doesn't happen.

**Comment by ilyavolodin** [comment]
This rule considers a variable as used when the variable value is either assigned to some other variable or used in the expression. Just assigning values to a variable doesn't mean that it's used somewhere.

**Comment by peduxe** [comment]
Ah okay, makes sense now, thanks.
